gateway-js/src/executeQueryPlan.ts

```diff
@@ -252,6 +257,42 @@ async function executeNode<TContext>(
   }
 }
 
+export function shouldSkipFetchNode(
+  node: FetchNode,
+  variables: VariableValues = {},
+) {
+  if (node.inclusionConditions) {
+    const shouldSkip = node.inclusionConditions.every((conditionals) => {
+      function resolveConditionalValue(conditional: 'skip' | 'include') {
+        const conditionalType = typeof conditionals[conditional];
+        if (conditionalType === 'boolean') {
+          return conditionals[conditional] as boolean;
+        } else if (conditionalType === 'string') {
+          return variables[conditionals[conditional] as string] as string;
+        } else {
+          return null;
+        }
+      }
+
+      const includeValue = resolveConditionalValue('include');
+      const skipValue = resolveConditionalValue('skip');
+
+      if (includeValue !== null && skipValue !== null) {
+        return !includeValue || skipValue;
+      } else if (includeValue !== null) {
+        return !includeValue;
+      } else if (skipValue !== null) {
+        return skipValue;
+      }
+
+      throw new Error('Programming error: unexpected value for conditional directive');
```

Comment:
I suspect there might be a way to write this more concisely. If I get the semantics right (I haven't tested it), would something like this work?

```suggestion
      if (includeValue !== null && !includeValue) {
        return true;
      }
      
     if (skipValue !== null && skipValue) {
        return true;
     }
     
     return false;
```

Reply:
Ah yeah, the case where both are provided doesn't actually need to be handled specifically. This considerably simplifies the code. With strict equality checks against boolean values we don't even have to check for null, just:
`return includeValue === false || skipValue === true;`

Resolved in c8fd47df